core/client/src/main/java/alluxio/client/file/policy/options/BlockLocationPolicyCreateOptions.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.file.policy.options;
+
+import alluxio.annotation.PublicApi;
+
+import com.google.common.base.Objects;
+
+/**
+ * Method options for creating a {@link BlockLocationPolicyCreateOptions}.
+ */
+@PublicApi
+public final class BlockLocationPolicyCreateOptions {
+  private String mLocationPolicyClassName;
+  private int mDeterministicHashPolicyNumShards;
+
+  /**
+   * @return the default {@link BlockLocationPolicyCreateOptions}
+   */
+  public static BlockLocationPolicyCreateOptions defaults() {
+    return new BlockLocationPolicyCreateOptions();
+  }
+
+  /**
+   * Creates a new instance with defaults.
+   */
+  private BlockLocationPolicyCreateOptions() {
+    mDeterministicHashPolicyNumShards = 1;
+  }
```

Comment:
I don't feel that is necessary in the current code as we are getting defaults from the property key in the InStreamOptions and OpenFileOptions.